src/odemis/driver/technolution.py

```diff
@@ -165,30 +354,30 @@ def __init__(self, name, role, parent, **kwargs):
         super(EBeamScanner, self).__init__(name, role, parent=parent, **kwargs)
 
         clockFrequencyData = self.parent.ASM_API_Get_Call("/scan/clock_frequency", 200)
-        # Check if clockFrequencyData holds the proper key
+        # Check if clockFrequencyData holds the proper keysa
         if 'frequency' not in clockFrequencyData:
             raise IOError("Could not obtain clock frequency, received data does not hold the proper key")
-        clockFrequency = clockFrequencyData['frequency']
-
-        self.clockPeriod = model.FloatVA(1 / clockFrequency, unit='s')
+        self.clockPeriod = model.FloatVA(1 / clockFrequencyData['frequency'], unit='s', readonly=True)
         self._shape = (6400, 6400)
         # The resolution min/maximum are derived from the effective cell size restriction defined in the API
         self.resolution = model.ResolutionVA((6400, 6400), ((10, 10), (1000 * 8, 1000 * 8)))
-        self.dwellTime = model.FloatContinuous(self.clockPeriod.value, (max(self.clockPeriod.value, 400e-9), 100.0),
+        self.dwellTime = model.FloatContinuous(self.clockPeriod.value, (max(self.clockPeriod.value, 4e-7), 1e-4),
                                                unit='s')
         self.pixelSize = model.TupleContinuous((4e-9, 4e-9), range=((1e-9, 1e-9), (1e-3, 1e-3)), unit='m',
                                                setter=self._setPixelSize)
         self.rotation = model.FloatContinuous(0, range=(0, 2 * numpy.pi), unit='rad')
-        self.scanFlyback = model.FloatContinuous(0, range=(0, 100), unit='s')
+
         self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V')
         self.scanGain = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V')
-        self.scanDelay = model.TupleContinuous((0, 0), range=((0, 0), (100000, 100000)), unit='s',
+
+        # TODO K.K. this unit's max cannot be in seconds! + the matching MPPC unit (checked in setter neither)
```

Comment:
(Outdated)
I think because the API does not provide a max this value was chosen to basically not have any limit present. However this changed.
Current implementation is:
        self.scanDelay = model.TupleContinuous((0.0, 0.0), range=((0.0, 0.0), (200e-6, 10.0)), unit='s',
                                               setter=self._setScanDelay)